azureadps-2.0/AzureAD/New-AzureADServiceAppRoleAssignment.md

```diff
@@ -18,21 +18,86 @@ New-AzureADServiceAppRoleAssignment -ObjectId <String> [-InformationAction <Acti
 ```
 
-{{ Add example description here }}
+In this example, a group is assigned an app role defined by a resource app. All users who are direct member of the assigned group are considered to be assigned the app role:
+
+- `ObjectId`:  The ObjectId of the app's service principal.
+- `ResourceId`: The ObjectId of the app's service principal.
+- `Id`: The Id of the app role (defined on the app's service principal) to assign to the group. If no app roles have been defined on resource app, you can use `00000000-0000-0000-0000-000000000000` to indicate the app is assigned to the group.
+- `PrincipalId`: The ObjectId of the group to which you are assigning the app role.
+
+> [!NOTE]
+> This example applies when `Connect-AzureAD` was called witout any parameters. See [Example 4](#example-4--when-connected-using-a-customer-owned-app-or-service-identity) to see how this cmdlet is used when connected using a customer-owned app registration or service identity. 
+
+### Example 4: When connected using a customer-owned app or service identity
+
+```powershell
+PS C:\> Connect-AzureAD -TenantId $tenantOrDomain -ApplicationId $appId -CertificateThumbprint $thumb
+PS C:\> New-AzureADServiceAppRoleAssignment -ObjectId $client.ObjectId -ResourceId $resource.ObjectId -Id $appRole.Id -PrincipalId $client.ObjectId
+```
+
+This cmdlet's behavior changes when connected to the Azure AD PowerShell module using a customer-owned app registration or service identity. This includes when [connecting as a service principal](Connect-AzureAD.md#example-3--connect-a-session-as-a-service-principal), and when using the `AadAccessToken` parameter with an access token obtained for a customer-owned app registration or service identity.
+
+Under these circumstances this cmdlet is only used for assigning an app role to another service principal, identified by the `ObjectId` and `PrincipalId` parameters:
+
+- `ObjectId`:  The ObjectId of the client service principal to which you are assigning the app role.
+- `ResourceId`: The ObjectId of the resource service principal (e.g. an API).
+- `Id`: The Id of the app role (defined on the resource service principal) to assign to the client service principal. If no app roles have been defined on the, you can use `00000000-0000-0000-0000-000000000000`.
+- `PrincipalId`: The ObjectId of the client service principal to which you are assigning the app role.
+
+When connected using a customer-owned app or service identity, use [New-AzureADUserAppRoleAssignment](New-AzureADUserAppRoleAssignment.md) and [New-AzureADGroupAppRoleAssignment](New-AzureADUserAppRoleAssignment.md) to create app role assignments to user and groups, respectively.
 
 ## PARAMETERS
 
 ### -Id
-Specifies the ID.
+Specifies the Id of the app role (defined on the resource service principal) to assign. If no app roles have been defined on the, you can use `00000000-0000-0000-0000-000000000000` to indicate assignment of the resource app or service, without specifying an app role.
```

Comment:
Missing word. "If no app roles have been defined on the...., you can use..." Please add.